browser/app/js/browser/MainActions.js

```diff
@@ -16,77 +16,107 @@
     let files = e.target.files
     let filesToUploadCount = files.length
     for (let i = 0; i < filesToUploadCount; i++) {
-      uploadFile(files.item(i))
+      func(files.item(i))
     }
     e.target.value = null
   }
 
-  const loggedIn = web.LoggedIn()
+  toggleAddNew() {
+    this.setState({
+      addNewActive: !this.state.addNewActive
+    })
+  }
+
+  openMakeBucket() {
+    this.setState({
+      makeBucketActive: true
+    })
+  }
+
+  closeMakeBucket() {
+    if (this.state.makeBucketActive) {
+      this.setState({
+        makeBucketActive: false
+      })
+    } else {
+      this.setState({
+        addNewActive: false
+      })
+    }
+  }
+
+  render() {
+    const { prefixWritable, uploadFile } = this.props
+    const loggedIn = web.LoggedIn()
+
+    if (loggedIn || prefixWritable) {
+      return (
+        <ClickOutHandler onClickOut={this.closeMakeBucket.bind(this)}>
+          <div
+            className={classNames({
+              "add-new": true,
+              "add-new--active": this.state.addNewActive,
+              "add-new--bucket": this.state.makeBucketActive
+            })}
+          >
+            <i
+              className="add-new__toggle"
+              onClick={this.toggleAddNew.bind(this)}
+            />
 
-  if (loggedIn || prefixWritable) {
-    return (
-      <Dropdown dropup className="feb-actions" id="fe-action-toggle">
-        <Dropdown.Toggle noCaret className="feba-toggle">
-          <span>
-            <i className="fa fa-plus" />
-          </span>
-        </Dropdown.Toggle>
-        <Dropdown.Menu>
-          <OverlayTrigger placement="left" overlay={uploadTooltip}>
-            <a href="#" className="feba-btn feba-upload">
+            <label
+              htmlFor="add-new-upload"
+              className="add-new__item add-new__item--upload"
+            >
               <input
                 type="file"
-                onChange={onFileUpload}
-                style={{ display: "none" }}
-                id="file-input"
+                onChange={this.onFileUpload.bind(this, uploadFile)}
+                id="add-new-upload"
                 multiple={true}
               />
-              <label htmlFor="file-input">
-                {" "}
-                <i className="fa fa-cloud-upload" />{" "}
-              </label>
-            </a>
-          </OverlayTrigger>
-          {loggedIn && (
-            <OverlayTrigger placement="left" overlay={makeBucketTooltip}>
-              <a
-                href="#"
+            </label>
+            {loggedIn && (
+              <div
                 id="show-make-bucket"
-                className="feba-btn feba-bucket"
-                onClick={e => {
-                  e.preventDefault()
-                  showMakeBucketModal()
-                }}
+                className="add-new__item add-new__item--bucket"
+                onClick={
+                  this.state.makeBucketActive
+                    ? ""
```

Comment:
I get the following warning when I click on the Make Bucket button. I think we can get rid of this error by replacing `""` with `undefined`.

```
Warning: Expected `onClick` listener to be a function, instead got a value of `string` type.
    in div (created by MainActions)
    in div (created by MainActions)
    in ClickOutComponent (created by MainActions)
    in MainActions (created by Connect(MainActions))
    in Connect(MainActions) (created by MainContent)
    in section (created by MainContent)
    in MainContent (created by Browser)
    in Browser (created by Connect(Browser))
    in Connect(Browser) (created by Route)
    in Route (created by App)
    in Switch (created by App)
    in App
    in Router
    in Provider
```

Reply:
Thank you for pointing out this. Will fix it.